front-end-components/src/components/cost-codes-list/component.tsx

```diff
@@ -0,0 +1,222 @@
+import { useContext } from "react";
+import { CostCodesListProps } from "src/components/cost-codes-list";
+import { CostCodeMapContext } from "src/contexts";
+
+export const CostCodesList: React.FC<CostCodesListProps> = (props) => {
+  const { category } = props;
+
+  const costCodeMap = useContext(CostCodeMapContext);
+
+  const getCostCodes = (
```

Comment:
To prevent duplication of the category-to-code(s) logic, would it be possible to change the `costCodeMap` sent to `front-end` from a `Dictionary<string, string>` to a `Dictionary<string, string[]>`, where the key is the category and the values are the codes? It looks like this wouldn't be an insignificant amount of work, but I'm thinking of what happens when new sub-categories come along or are re-parented.

Reply:
Ah, I see. How irritating. This goes back to our conversation around chart IDs I guess. Maybe leave as-is for now to get this in, but it would be good to get a follow-up work item to look at again.

Reply:
As per discussion, leave logic as-is for the time being, but move `useContext(CostCodeMapContext)` into a new hook in `/src/contexts/hooks.tsx` that returns the `getCostCodes()` function directly, keeping the internals out of the `<CostCodesList />` component.